### ISSUE TITLE
tests/drivers/ipm/peripheral.mailbox failing sporadically on qemu_x86_64 (timeout)

### ISSUE THREAD

**Issue by nashif** [zephyrproject-rtos/zephyr#12478]
see https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/31042/6/tests, the test times out with

```
***** Booting Zephyr OS zephyr-v1.13.0-3219-gdfc8f0a6d5 *****
starting test - Test IPM
ipm_recv0: 'everything is awesome'
ipm_recv0: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor '

```

**Comment by andyross** [comment]
OK, confirmed.  I got 3 failures out of 10 tries running this test under load on my CI (-ish) docker instance.  Interestingly not one failure in 40 running the same binaries built with the same SDK under the host qemu.

So... plausibly timing related.  Could be as simple as a stack overflow that depends on when the interrupt lands, if I'm lucky.

**Comment by wentongwu** [comment]
tried so many times with script to reproduce this issue, but failed to reproduce it on my Ubuntu18.04 PC so far.  talked with @andyross, I will try to reproduce it with docker. If anyone can easily reproduce it, please tell me... thanks

**Comment by wentongwu** [comment]
with https://docs.docker.com/install/linux/docker-ce/ubuntu/, didn't setup docker successfully(welcome setup docker's BKM). So try to reproduce it with Cinly's Fedora PC(Commit SHA: g4278bf567348).

test it with below command  for i in {1..100}; do scripts/sanitycheck -T tests/drivers/ipm/ -p qemu_x86_64; done (disable the exclude #platform_exclude: qemu_x86_64 in testcase.yaml)

when open SMP in soc config, this issue can reproduce. with SMP disabled, no fail occurred. So I'm checking SMP's code...

**Comment by wentongwu** [comment]
this issue has already submitted PR(#14619 ), but this PR depends on https://github.com/zephyrproject-rtos/zephyr/issues/14807.

**Comment by andyross** [comment]
Took a look here.  As far as I can tell, the earlier problem has disappeared and the only issue is, as @wentongwu discovered, SMP correctness.  There's no point in trying to finesse that with a controversial patch at this stage.  We have **lots** of known SMP safety issues in the test code already, and the workaround is just to disable SMP on those tests.  We can come back in 1.15 as part of the work for #14639 and fix things at our leisure.

**Comment by nashif** [comment]
problem still exist

```
c[?7l[2J[0mSeaBIOS (version rel-1.12.0-0-ga698c8995f-prebuilt.qemu.org)
Booting from ROM..***** Booting Zephyr OS zephyr-v1.14.0-3-g9a8467d83aaf *****
starting test - Test IPM
ipm_recv0: 'everything is awesome'
ipm_recv0: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor '
ipm_recv0: 'incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nost'
ipm_recv0:ASSERTION FAIL [z_spin_lock_valid(l)] @ ../../../../../../include/spinlock.h:78
	Recursive spinlock
*** FATAL ERROR vector 6661 code 1016
***  RIP 9:0x100e46 RSP 16:0x105ba8 RFLAGS 0x2
***  RAX 0x105327 RCX 0xa RDX 0x3f8 RSI 0x4e RDI 0x1052d9
***  R8 0xffffffff R9 0x105ac8 R10 0x0 R11 0x0
```

**Comment by rljordan** [comment]
Can you please retest this?  Might be fixed.

**Comment by wentongwu** [comment]
With 12 hours stress test, can’t reproduce.

**Comment by rljordan** [comment]
Anas can we close this?

**Comment by jenmwms** [comment]
Please note that if we close this, we need to revisit the commit 0dd0f9f52a10f1bee71e21ccc7daba7c59b89dbf to enable (it was disabled in the commit to allow testing).

**Comment by wentongwu** [comment]
The case is really not designed for SMP as PR #14619 found, and I will disable SMP for this case, but I will do the change as commit https://github.com/zephyrproject-rtos/zephyr/pull/18531/commits/ab3a20921d11392be911861d448126cae72da614 do.

**Comment by nashif** [comment]
test not supporting SMP is not a medium bug, lowering priority

**Comment by pfalcon** [comment]
Hello again. We have tests/drivers/ipm/ failing reproducibly as part of https://github.com/zephyrproject-rtos/zephyr/pull/23693 , with a stack overflow. I proceeded to check if bumping stack would resolve it. Of course, I started with trying that with master (7ca6a05159e9dfdce3e257ec4fc4d5b5b7ad3127) first (Zephyr SDK 0.11.2). 

I found that 3 out of 3 initial run hanged with:
~~~
[QEMU] CPU: qemu64,+x2apic
qemu-system-x86_64: warning: TCG doesn't support requested feature: CPUID.01H:ECX.x2apic [bit 21]
SeaBIOS (version rel-1.12.1-0-ga5cab58-dirty-20200214_052440-f7294c49af13-zephyr
)
Booting from ROM..Optimal CONFIG_X86_MMU_PAGE_POOL_PAGES 8
*** Booting Zephyr OS build zephyr-v2.2.0-1489-g7ca6a05159e9  ***
starting test - Test IPM
ipm_recv0: 'everything is awesome'
ipm_recv0: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor '
ipm_recv0: 'incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nost'
QEMU: Terminated
~~~

Then, in next runs, can't reproduce any more. So, initially reopened, but closing again.
